frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1240,6 +1241,14 @@ export const dashboardLogic = kea<dashboardLogicType>([
                         return true
                     }
                 })
+                // sort tiles by layout to ensure insights are computed in order of appearance on dashboard
+                .sort((a, b) => {
+                    const ay = a.layouts?.xs?.y ?? 0
+                    const ax = a.layouts?.xs?.x ?? 0
+                    const by = b.layouts?.xs?.y ?? 0
+                    const bx = b.layouts?.xs?.x ?? 0
+                    return ay !== by ? ay - by : ax - bx
```

Comment:
This only considers the `xs` layout for ordering the tiles. It would be better if we'd consider the currently active layout. Not a blocker though.

Reply:
spoke on slack, there's not a way to get active layout on the backend right now. But important thing now is that both backend and frontend use the same sorting logic so that the frontend polls for insight in the same way as backend is calculating them. (drawback of this change)

Have also added a feature flag on the backend chaining so can do a phased rollout and test this out.